lib/Dialect/FIRRTL/FIRRTLAnnotations.cpp

```diff
@@ -17,11 +17,32 @@ using namespace circt;
 using namespace firrtl;
 
 static ArrayAttr getAnnotationsFrom(Operation *op) {
-  if (auto annots = op->getAttrOfType<ArrayAttr>("annotations"))
+  if (auto annots = op->getAttrOfType<ArrayAttr>(getAnnotationAttrName()))
     return annots;
   return ArrayAttr::get(op->getContext(), {});
 }
 
+static ArrayAttr getAnnotationsFrom(ArrayRef<Annotation> annotations,
+                                    MLIRContext *context) {
+  if (annotations.empty())
+    return ArrayAttr::get(context, {});
+  SmallVector<Attribute, 8> attrs;
```

Comment:
Is there a good reason to choose `8` as opposed to just `SmallVector<Attribute>`?

Reply:
Nah good point, I'll just use the default here.